airflow/models/dag.py

```diff
@@ -2020,6 +2025,15 @@ def create_dagrun(
                 "Creating DagRun needs either `run_id` or both `run_type` and `execution_date`"
             )
 
+        try:
+            for k, v in copy.deepcopy(self.params).items():
+                if isinstance(v, Param):
+                    if conf and k in conf:
+                        v.default = conf[k]
+                    v()
```

Comment:
We should delay the deepcopy until as late as possible
```suggestion
            for k, v in self.params.items():
                if isinstance(v, Param):
                    if conf and k in conf:
                        v = copy.deepcopy(v)
                        v.default = conf[k]
                    v()
```

Reply:
Actually -- we shouldn't ever change v.default -- this seems like an odd pattern to set overwrite the default with the provided value.